src/main/java/io/github/nucleuspowered/nucleus/modules/fun/commands/LightningCommand.java

```diff
@@ -0,0 +1,103 @@
+import org.spongepowered.api.event.cause.Cause;
+import org.spongepowered.api.event.cause.NamedCause;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.util.blockray.BlockRay;
+import org.spongepowered.api.util.blockray.BlockRayHit;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@RegisterCommand({"lightning", "smite", "thor"})
+public class LightningCommand extends CommandBase<CommandSource> {
+
+    private final String player = "player";
+
+    @Override
+    public Map<String, PermissionInformation> permissionSuffixesToRegister() {
+        Map<String, PermissionInformation> m = new HashMap<>();
+        m.put("others", new PermissionInformation(Util.getMessageWithFormat("permission.others", this.getAliases()[0]), SuggestedLevel.ADMIN));
+        return m;
+    }
+
+    @Override
+    public CommandSpec createSpec() {
+        return CommandSpec.builder().executor(this).arguments(GenericArguments.optional(GenericArguments.requiringPermission(
+                GenericArguments.onlyOne(GenericArguments.player(Text.of(player))), permissions.getPermissionWithSuffix("others")))).build();
+    }
+
+    @Override
+    public CommandResult executeCommand(final CommandSource src, CommandContext args) throws Exception {
+        Optional<Player> opl = this.getUser(Player.class, src, player, args);
+        if (!opl.isPresent()) {
+            return CommandResult.empty();
+        }
+
+        Player pl = opl.get();
+
+        if (pl == src) {
+            BlockRay<World> playerBlockRay = BlockRay.from(pl).blockLimit(350).build();
+            BlockRayHit<World> finalHitRay = null;
+
+            while (playerBlockRay.hasNext()) {
+                BlockRayHit<World> currentHitRay = playerBlockRay.next();
+
+                if (!pl.getWorld().getBlockType(currentHitRay.getBlockPosition()).equals(BlockTypes.AIR)) {
+                    finalHitRay = currentHitRay;
+                    break;
+                }
+            }
+
+            Location<World> lightningLocation = null;
+
+            if (finalHitRay == null) {
+                lightningLocation = pl.getLocation();
+            } else {
+                lightningLocation = finalHitRay.getLocation();
+            }
+
+            Text message = this.spawnLightning(lightningLocation, src) ? Util.getTextMessageWithFormat("command.lightning.success")
+                    : Util.getTextMessageWithFormat("command.lightning.error");
+            pl.sendMessage(message);
+        } else {
+            Text message = this.spawnLightning(pl.getLocation(), src) ? Util.getTextMessageWithFormat("command.lightning.success.other", pl.getName())
+                    : Util.getTextMessageWithFormat("command.lightning.error");
+            pl.sendMessage(message);
+        }
+
+        return CommandResult.success();
```

Comment:
Some scenarios that end in an error condition get here. They should never be marked as a success.